x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,2862 @@
+    - differ between base and current, but doesn't update target, **keep current** version. (A B A)
+    - differ between base and current, updates target, **keep current** version. (A B B)
+    - differ between all version (base, current and target), **keep current** version. (A B C)
+  - If element index exists in base and current version, but not in target:
+    - If base differs from current: **keep current** version. (A B _)
+    - If base equals current: **remove element**.  (A A _)
+  - If element index exists in base and target version, but not in current:
+    - **Remove element**. (A _ B) and (A _ A)
+  - If element index exists in current and target version, but not in base:
+    - If base equals current: **add element/keep target**. (_ A A)
+    - If base differs from current: **keep both current and target**. (_ A B)
+  - If element index exists only in base version:
+    - **Remove element** (A _ _)
+  - If element index exists only in current version:
+    - **Keep current version** (_ A _)
+  - If element index exists only in target version:
+    - **Keep target version** (_ _ A)
+    
+<table>
+    <thead>
+      <tr>
+        <th style="border-right:3px solid black">Use case</th>
+        <th>Base version</th>
+        <th>Current version</th>
+        <th style="border-right:3px solid black">Target version</th>
+        <th>Merged version (output)</th>
+      </tr>
+    </thead>
+  <tbody valign="top">
+    <tr>
+      <td rowspan=2 style="border-right:3px solid black">Keep user changes if element conflicts, but apply changes in elements with clean updates <br><br> <ul><li><b>1st elem: (A B C - keep current) </li><li>2nd elem: (A A B - keep target)</b></li></ul><br><br> Same for: <br><ul><li><b>1st elem: (A B C) + 2nd elem: (A B A - keep current)</b></li><li><b>1st elem: (A B C) + 2nd elem: (A B B - keep current/target)</b></li></ul></td>
+      <td>
+      <pre>
+[
+  logs-*,
+  logstash-*
+]
+        </pre>
+      </td>
+      <td>
+        <pre>
+[
+  cluster_one:logs-*,
+  cluster_one:logstash-*
+]
+        </pre>
+      </td>
+      <td style="border-right:3px solid black">
+      <pre>
+[
+  new-index1-*,
+  new-index2-*,
+]
+        </pre>
+        </td>
+      <td>
+      <pre>
+[
+  cluster_one:logs-*,
+  cluster_one:logstash-*
+]
+        </pre>
+      </td>
+    </tr>
+    <tr>
+      <td>
+      <pre>
+[
+  {
+    ecs: true,
+    name: "event.action",
+    type: "unknown"
+  },
+  {
+    ecs: false,
+    name: "host.os.type",
+    type: "keyword"
+  },
+]
+        </pre>
+      </td>
+      <td>
+        <pre>
+[
+  {
+    ecs: true,
+    name: "event.UPDATED",
+    type: "unknown"
+  },
+  {
+    ecs: false,
+    name: "host.os.type",
+    type: "keyword"
+  },
+]
+        </pre>
+      </td>
+      <td style="border-right:3px solid black">
+      <pre>
+[
+  {
+    ecs: true,
+    name: "event.action",
+    type: "keyword"
+  },
```

Comment:
I think this situation should end up with a conflict, because:

- In the current version, `event.action` was replaced by `event.UPDATED`, in other words `event.action` was removed and `event.UPDATED` was added.
- In the target version, `event.action` was updated: its `type` was changed from `unknown` to `keyword`.

It shouldn't be possible to simultaneously remove and update the same element.

What should be in the merged version in this case? Probably just the current version.